client/beefy/src/import.rs

```diff
@@ -151,42 +122,36 @@ where
 		let hash = block.post_hash();
 		let number = *block.header.number();
 
-		let beefy_proof = block
-			.justifications
-			.as_mut()
-			.and_then(|just| {
-				let decoded = just
-					.get(BEEFY_ENGINE_ID)
-					.map(|encoded| self.decode_and_verify(encoded, number, hash));
-				// Remove BEEFY justification from the list before giving to `inner`;
-				// we will append it to backend ourselves at the end if all goes well.
-				just.remove(BEEFY_ENGINE_ID);
-				decoded
-			})
-			.transpose()
-			.unwrap_or(None);
+		let beefy_encoded = block.justifications.as_mut().and_then(|just| {
+			let encoded = just.get(BEEFY_ENGINE_ID).cloned();
+			// Remove BEEFY justification from the list before giving to `inner`;
+			// we will append it to backend ourselves at the end if all goes well.
+			just.remove(BEEFY_ENGINE_ID);
```

Comment:
Not a part of this PR. but probably let's change `remove` to return removed justification? Then we'll need to only call `remove` instead of `get` and `remove`?